deepchem/datasets/__init__.py

```diff
@@ -682,6 +682,40 @@ def _update_mean_and_std(self, df):
       save_to_disk(ys, os.path.join(self.data_dir, row['y_sums']))
       save_to_disk(yss, os.path.join(self.data_dir, row['y_sum_squares']))
 
+  def get_grad_statistics(self):
+    """Computes and returns statistics of this dataset
+
+    This function assumes that the first task of a dataset holds the energy for
+    an input system, and that the remaining tasks holds the gradient for the
+    system.
+
+    TODO(rbharath, joegomes): It is unclear whether this should be a Dataset
```

Comment:
I agree, I initially added this here in the spirit of keeping everything that access the metadata_df in the dataset object but currently would only be used by PES fitting